flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobCacheSizeTracker.java

```diff
@@ -99,44 +101,56 @@ public BlobCacheSizeTracker(long sizeLimit) {
     }
 
     /** Register the target file to the tracker. */
-    public void track(@Nullable JobID jobId, BlobKey blobKey, long size) {
+    public void track(JobID jobId, BlobKey blobKey, long size) {
+        checkNotNull(jobId);
         checkNotNull(blobKey);
         checkArgument(size >= 0);
 
         synchronized (lock) {
-            caches.put(Tuple2.of(jobId, blobKey), size);
-            if (jobId != null) {
+            if (caches.putIfAbsent(Tuple2.of(jobId, blobKey), size) == null) {
                 blobKeyByJob.computeIfAbsent(jobId, ignore -> new HashSet<>()).add(blobKey);
-            }
-            total += size;
-            if (total > sizeLimit) {
+
+                total += size;
+                if (total > sizeLimit) {
+                    LOG.warn(
+                            "The overall size of ShuffleDescriptors in PermanentBlobCache exceeds "
```

Comment:
`ShuffleDescriptors` -> `blob`

Reply:
Thank you for pointing this out. Resolved.